sql/aiven_extras.sql

```diff
@@ -453,6 +453,46 @@ END;
 $$;
 
 
+DROP FUNCTION IF EXISTS aiven_extras.set_pgaudit_parameter(TEXT, TEXT, TEXT);
+CREATE FUNCTION aiven_extras.set_pgaudit_parameter(
+    arg_parameter TEXT,
+    arg_database TEXT,
+    arg_value TEXT
+)
+RETURNS VOID LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = pg_catalog, aiven_extras
+AS $$
+BEGIN
+    IF (
+        SELECT usesuper
+            FROM pg_catalog.pg_database d
+                JOIN pg_user u
+                    ON (u.usesysid = d.datdba)
+                WHERE d.datname = arg_database
+                LIMIT 1
+    ) THEN
+        RAISE EXCEPTION 'Cannot modify superuser-owned database: %', arg_database;
+    ELSIF arg_parameter NOT IN (
+        'log',
+        'log_catalog',
+        'log_parameter',
+        'log_relation',
+        'log_statement',
+        'log_statement_once'
+    ) THEN
+        RAISE EXCEPTION 'Invalid parameter: %', arg_parameter;
+    END IF;
+
+    EXECUTE format('ALTER DATABASE %I SET pgaudit.%s = %L',
```

Comment:
changing the `%s` to `%I` would be an optional hardening.

This shouldn't be reachable as malformed/injectable code due to the `Invalid parameter` check, however should this check change in the future, we would already be covered.

Reply:
I'm unsure if `pgaudit."identifier"` would work. Mabye `EXECUTE format(' ... SET %I ...', ..., 'pgaudit.' || arg_parameter , ...)`?

Reply:
It won't quote unless there is a space or quote present; keeping behaviour as is. It just covers the case where the parameter check is bypassed and something like `log; ALTER USER x WITH SUPERUSER; --` is supplied. With `%I` , which would then end-up as `ALTER DATABASE mydb SET pgaudit."log'=1; ALTER USER sushi WITH SUPERUSER; --" = 'c'`. This works better than `%L` which will quote everything; breaking syntax 
```
LINE 1: ALTER DATABASE blah SET pgaudit.'log' = 'a'
                                        ^
QUERY:  ALTER DATABASE blah SET pgaudit.'log' = 'a'

```

Reply:
Thanks for the input! I updated the PR with the suggestions.